cgi-bin/LJ/Userpic.pm

```diff
@@ -524,27 +548,11 @@ sub keywords {
 
 sub imagedata {
     my $self = $_[0];
-
     $self->load_row or return undef;
-    my $u = $self->owner;
-
     return undef if $self->expunged;
 
-    # check mogile
-    if ( $self->in_mogile ) {
-        my $key = $u->mogfs_userpic_key( $self->picid );
-        my $data = LJ::mogclient()->get_file_data( $key );
-        return $$data;
-    }
-
-    # check userpicblob2 table
-    my $dbb = LJ::get_cluster_reader($u)
-        or return undef;
-
-    my $data = $dbb->selectrow_array( "SELECT imagedata FROM userpicblob2 WHERE ".
-                                      "userid=? AND picid=?", undef, $self->userid,
-                                      $self->picid );
-    return $data ? $data : undef;
```

Comment:
So I just discovered another problem - this is dropping backward compatibility for retrieving icons stored in userpicblob2, which will include any site which wasn't previously running MogileFS (which is most of them because it was too hard to set up).  I didn't notice this immediately because of my browser cache.  We need to either keep this fallback, or include a process for migrating these images out of userpicblob2 when upgrading.